Изменения №1 от 20.04.2023

ZhosanDenis opened the pull request:
- создан класс-сущность Client для записи статуса клиента (интересует приют для кошек или для собак);
- создан ClientRepository для добавления сущностей в БД;
- в enum-класс Command добавлены команды по выбору приюта для кошек и приюта для собак, команда по запросу контактных данных службы охраны. Также переименованы команды с DOG в названии (теперь PET);
- добавлена схема проезда для приюта для собак (предыдущая схема проезда используется для приюта для кошек);
- в scripts.sql добавлен changeset с созданием таблицы clients;
- в TelegramBotConfiguration переименованы команды согласно изменениям в enum-классе Command;
- TelegramBotUpdatesListener: 
1) заинжектен ClientRepository;
2) изменены обработки сообщений от опекунов и из чата волонтеров (теперь каждые сообщения/команды обрабатываются в разных методах в зависимости откуда отправлены); 
3) изменен метод по созданию/обновлению чата волонтеров (теперь чат создается в зависимости от нажатой кнопки в чате волонтеров); 
4) изменена обработка сообщений от опекуна в зависимости от того, информацию по какому приюту он ищет; 
5) создан метод по обработке команд/сообщений из чата волонтеров; 
6) в методе по принятию решения об окончании испытательного срока конструкция if-else заменена на switch-case; 
7) обработка команды по уведомлению опекуна о некачественном заполнении отчетов перенесена в метод по обработке сообщений из чата волонтеров; 
8) добавлена клавиатура с выбором приюта;
9) в методе с созданием клавиатуры "Как взять питомца из приюта" (этап 2 из ТЗ) теперь создается две разных клавиатуры в зависимости от типа приюта.

ZhosanDenis committed:
Изменения №1 от 20.04.2023
- создан класс-сущность Client для записи статуса клиента (интересует приют для кошек или для собак);
- создан ClientRepository для добавления сущностей в БД;
- в enum-класс Command добавлены команды по выбору приюта для кошек и приюта для собак, команда по запросу контактных данных службы охраны. Также переименованы команды с DOG в названии (теперь PET);
- добавлена схема проезда для приюта для собак (предыдущая схема проезда используется для приюта для кошек);
- в scripts.sql добавлен changeset с созданием таблицы clients;
- в TelegramBotConfiguration переименованы команды согласно изменениям в enum-классе Command;
- TelegramBotUpdatesListener:
1) заинжектен ClientRepository;
2) изменены обработки сообщений от опекунов и из чата волонтеров (теперь каждые сообщения/команды обрабатываются в разных методах в зависимости откуда отправлены);
3) изменен метод по созданию/обновлению чата волонтеров (теперь чат создается в зависимости от нажатой кнопки в чате волонтеров);
4) изменена обработка сообщений от опекуна в зависимости от того, информацию по какому приюту он ищет;
5) создан метод по обработке команд/сообщений из чата волонтеров;
6) в методе по принятию решения об окончании испытательного срока конструкция if-else заменена на switch-case;
7) обработка команды по уведомлению опекуна о некачественном заполнении отчетов перенесена в метод по обработке сообщений из чата волонтеров;
8) добавлена клавиатура с выбором приюта;
9) в методе с созданием клавиатуры "Как взять питомца из приюта" (этап 2 из ТЗ) теперь создается две разных клавиатуры в зависимости от типа приюта.

ZhosanDenis committed:
Изменения №1 от 21.04.2023
- созданы сущности с припиской Cat и Dog, сущности без приписок стали родительскими классами для созданных сущностей;
- для каждой сущности с припиской создан репозиторий;
- сущности с приписками перенесены в отдельный пакет в зависимости от типа животного;
- в некоторые эндпоинты контроллеров добавлены параметры с типом животного для получения нужных данных;
- в классы-record добавлено поле petType;
- в классах-сервисах возвращаемый тип данных зависит от типа животного, закомментированы (временно) методы для работы с телеграм ботом;
- созданы таблицы в БД для сущностей Cat, переименован таблицы для сущностей Dog;
- закомментирован (временно) код в TelegramBotUpdatesListener и Timer для проверки работы контоллеров-сервисов.

ZhosanDenis committed:
Изменения №1 от 23.04.2023
- AgreementController: добавлен эндпоинт по изменению даты окончания испытательного срока (для варианта возврата питомца раньше);
- AgreementService: изменен метод по получению всех договоров с действующим испытательным сроком (в зависимости от типа животного), изменен метод по получению всех договоров с заканчивающимся (осталось 2 дня до конца) испытательным сроком (в зависимости от типа животного);
- application.properties: убран параметр, показывающий действия Hibernate;
- CarerService: изменен метод по добавлению опекуна (из телеграм бота) - в зависимости от типа животного, изменен метод по поиску опекуна по id чата (из телеграм бота) - в зависимости от типа животного, удален неиспользующийся метод по поиску опекуна по id питомца;
- DailyReport: добавлено поле Carer carer + геттер/сеттер, переименованы поля с dog на pet;
- DailyReportRecord: переименованы поля с dog на pet;
- DailyReportService: изменен метод по добавлению ежедневного отчета из телеграм бота (в зависимости от типа животного);
- ModelMapper: в методе updateCarer() повторяющийся код вынесен в отдельный private метод update(), изменен метод mapToDailyRecordRecord() для правильного сохранения ежедневного отчета;
- Pet: добавлены public константы для использования на проверках типа животного по всему приложению;
- TelegramBotConfiguration: удалены команды, связанные с кинологами, чтобы опекуны кошек их не видели;
- TelegramBotUpdatesListener: заинжектены CatCarerRepository и CatAgreementRepository, изменен метод по сохранению фотографии и созданию ежедневного отчета (в зависимости от типа животного), добавлен метод selectVolunteerChatId для выбора чата волонтеров, для решения об окончании испытательного срока (метод decideOnProbation()) предусмотрены обработки команд в зависимости от типа животного, в методе handleTextMessage() изменена логика сохранения частей ежедневного отчета;
- Timer: теперь таймер ищет информацию о договорах в обеих таблицах БД (dog_agreements и cat_agreements) и отправляет напоминание о проверке отчетов в оба чата волонтеров (для кошек и собак).

ZhosanDenis committed:
Изменения №2 от 23.04.2023
- подкорректировал DogControllerTest: добавил необходимые параметры, изменил ссылки запросов. Тесты работают.

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/AgreementController.java:
```diff
@@ -41,8 +44,9 @@ public ResponseEntity<AgreementRecord> addAgreement(
-            @PathVariable Long id) {
-        return ResponseEntity.ok(agreementService.findAgreementById(id));
+            @PathVariable Long id,
+            @RequestParam(defaultValue = "кошка/собака") String petType) {
```
а что за значение такое дефолтное? и лучше принимать enum, а не строку

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/AgreementController.java:
```diff
@@ -61,8 +65,9 @@ public ResponseEntity<AgreementRecord> editAgreement(
-            @PathVariable long id) {
-        agreementService.deleteAgreement(id);
+            @PathVariable long id,
+            @RequestParam(defaultValue = "кошка/собака") String petType) {
```
а что за значение такое дефолтное? и лучше принимать enum, а не строку

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/AgreementController.java:
```diff
@@ -71,8 +76,20 @@ public ResponseEntity<?> deleteAgreementById(
     @GetMapping
-    public ResponseEntity<List<AgreementRecord>> findAllAgreements() {
-        return ResponseEntity.ok(this.agreementService.findAllAgreements());
+    public ResponseEntity<List<AgreementRecord>> findAllAgreements(@RequestParam(defaultValue = "кошка/собака") String petType) {
```
а что за значение такое дефолтное? и лучше принимать enum, а не строку

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/AgreementController.java:
```diff
@@ -71,8 +76,20 @@ public ResponseEntity<?> deleteAgreementById(
+    @PatchMapping("/{id}")
+    public ResponseEntity<AgreementRecord> changeProbationEndData(@PathVariable long id,
+                                                                  @RequestParam LocalDate localDate,
+                                                                  @RequestParam(defaultValue = "кошка/собака") String petType) {
```
а что за значение такое дефолтное? и лучше принимать enum, а не строку

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/CarerController.java:
```diff
@@ -38,39 +38,41 @@ public CarerController(CarerService carerService) {
-            @PathVariable long id) {
-        return ResponseEntity.ok(this.carerService.findCarer(id));
+            @PathVariable long id,
+            @RequestParam(defaultValue = "кошка/собака") String petType) {
```
а что за значение такое дефолтное? и лучше принимать enum, а не строку

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/CarerController.java:
```diff
@@ -93,15 +95,16 @@ public ResponseEntity<CarerRecord> editCarer(
-            @PathVariable long id) {
-        this.carerService.deleteCarer(id);
+            @PathVariable long id,
+            @RequestParam(defaultValue = "кошка/собака") String petType) {
```
а что за значение такое дефолтное? и лучше принимать enum, а не строку

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/CarerController.java:
```diff
@@ -112,17 +115,18 @@ public ResponseEntity<?> deleteCarer(
-            @RequestParam String phone) {
-        return ResponseEntity.ok(this.carerService.findCarerByPhoneNumber(phone));
+            @RequestParam String phone,
+            @RequestParam(defaultValue = "кошка/собака") String petType) {
```
а что за значение такое дефолтное? и лучше принимать enum, а не строку

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/CarerController.java:
```diff
@@ -112,17 +115,18 @@ public ResponseEntity<?> deleteCarer(
     @GetMapping
-    public ResponseEntity<List<CarerRecord>> findAllCarers() {
-        return ResponseEntity.ok(this.carerService.findAllCarers());
+    public ResponseEntity<List<CarerRecord>> findAllCarers(@RequestParam(defaultValue = "кошка/собака") String petType) {
```
а что за значение такое дефолтное? и лучше принимать enum, а не строку

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/DailyReportController.java:
```diff
@@ -57,9 +58,10 @@ public DailyReportController(DailyReportService dailyReportService) {
             @Parameter(description = "ID опекуна", example = "1")
-            @RequestParam(name = "Идентификатор опекуна") Long carerId) {
+            @RequestParam(name = "Идентификатор опекуна") Long carerId,
+            @RequestParam(defaultValue = "кошка/собака") String petType) {
```
а что за значение такое дефолтное? и лучше принимать enum, а не строку

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/DailyReportController.java:
```diff
@@ -83,9 +85,10 @@ public ResponseEntity<List<DailyReportRecord>> findDailyReportsByCarerId(
             @RequestParam Long carerId,
-            @RequestParam LocalDate reportDate) {
+            @RequestParam LocalDate reportDate,
+            @RequestParam(defaultValue = "кошка/собака") String petType) {
```
а что за значение такое дефолтное? и лучше принимать enum, а не строку

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/DailyReportController.java:
```diff
@@ -96,9 +99,10 @@ public ResponseEntity<DailyReportRecord> findDailyReportsByCarerAndDate(
             @Parameter(description = "Дата отчета", example = "2023-01-01")
-            @RequestParam(name = "Дата отчета") LocalDate localDate) {
+            @RequestParam(name = "Дата отчета") LocalDate localDate,
+            @RequestParam(defaultValue = "кошка/собака") String petType) {
```
а что за значение такое дефолтное? и лучше принимать enum, а не строку

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/DailyReportController.java:
```diff
@@ -107,9 +111,12 @@ public ResponseEntity<List<DailyReportRecord>> findDailyReportsByDate(
-    public void downloadPhotoByByCarerIdAndDate(long carerId, LocalDate reportDate, HttpServletResponse response) {
+    public void downloadPhotoByByCarerIdAndDate(@RequestParam long carerId,
+                                                @RequestParam LocalDate reportDate,
+                                                @RequestParam(defaultValue = "кошка/собака") String petType,
```
а что за значение такое дефолтное? и лучше принимать enum, а не строку

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/DailyReportController.java:
```diff
@@ -123,13 +130,15 @@ public void downloadPhotoByByCarerIdAndDate(long carerId, LocalDate reportDate,
-    public ResponseEntity<?> deleteDailyReport(@PathVariable long id) {
-        this.dailyReportService.deleteDailyReport(id);
+    public ResponseEntity<?> deleteDailyReport(@PathVariable long id,
+                                               @RequestParam(defaultValue = "кошка/собака") String petType) {
```
а что за значение такое дефолтное? и лучше принимать enum, а не строку

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/PetController.java:
```diff
@@ -71,22 +82,24 @@ public ResponseEntity<DogRecord> addDog(@RequestBody DogRecord dogRecord) {
-    public ResponseEntity<DogRecord> findDog(@Parameter(description = "Введите ID Собаки") @PathVariable long id) {
-        return ResponseEntity.ok(this.dogService.findDog(id));
+    public ResponseEntity<PetRecord> findDog(@Parameter(description = "Введите ID питомца") @PathVariable long id,
+                                             @RequestParam(defaultValue = "кошка/собака") String petType) {
```
а что за значение такое дефолтное? и лучше принимать enum, а не строку

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/PetController.java:
```diff
@@ -97,60 +110,63 @@ public ResponseEntity<DogRecord> findDog(@Parameter(description = "Введит
-        this.dogService.deleteDog(id);
+    public ResponseEntity<?> deleteDog(@Parameter(description = " Введите ID питомца для удаления")
+                                       @PathVariable long id,
+                                       @RequestParam(defaultValue = "кошка/собака") String petType) {
```
а что за значение такое дефолтное? и лучше принимать enum, а не строку

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/PetController.java:
```diff
@@ -97,60 +110,63 @@ public ResponseEntity<DogRecord> findDog(@Parameter(description = "Введит
     @GetMapping
-    public ResponseEntity<List<DogRecord>> findAllDogs() {
-        return ResponseEntity.ok(this.dogService.findAllDogs());
+    public ResponseEntity<List<PetRecord>> findAllPets(@RequestParam(defaultValue = "кошка/собака") String petType) {
```
а что за значение такое дефолтное? и лучше принимать enum, а не строку

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/PetController.java:
```diff
@@ -97,60 +110,63 @@ public ResponseEntity<DogRecord> findDog(@Parameter(description = "Введит
     )
     @PatchMapping("/is-taken/{id}")
     public ResponseEntity<?> changeIsTakenStatus(@PathVariable long id,
+                                                 @RequestParam(defaultValue = "кошка/собака") String petType,
```
а что за значение такое дефолтное? и лучше принимать enum, а не строку

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/PetController.java:
```diff
@@ -97,60 +110,63 @@ public ResponseEntity<DogRecord> findDog(@Parameter(description = "Введит
+            tags = "Питомец")
     @PatchMapping("/on-probation/{id}")
     public ResponseEntity<?> changeOnProbationStatus(@PathVariable long id,
+                                                     @RequestParam(defaultValue = "кошка/собака") String petType,
```
а что за значение такое дефолтное? и лучше принимать enum, а не строку

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/listener/TelegramBotUpdatesListener.java:
```diff
@@ -127,29 +155,24 @@ public int process(List<Update> updates) {
-                                    .findByTelegramChatId(volunteerChatId);
-                            updateVolunteerChat(newVolunteerChatId, volunteerChat);
-                        }
+                    } else if (chatType.equals("group") && update.message() != null && update.callbackQuery() == null) {
```
лучше сделать enum-ом тип чата - парсить и проверять по enum

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/listener/TelegramBotUpdatesListener.java:
```diff
@@ -159,9 +182,14 @@ public int process(List<Update> updates) {
-        volunteerChat.setName("Чат волонтеров " + LocalDate.now());
+    private void updateVolunteerChat(long newVolunteerChatId, VolunteerChat volunteerChat, long volunteersId) {
+        if (volunteersId == 1) {
+            volunteerChat.setId(1);
```
не надо Id вручную ставить никогда

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/listener/TelegramBotUpdatesListener.java:
```diff
@@ -159,9 +182,14 @@ public int process(List<Update> updates) {
+            volunteerChat.setId(1);
+            volunteerChat.setName("Чат волонтеров приюта для кошек " + LocalDate.now());
+        } else if (volunteersId == 2) {
+            volunteerChat.setId(2);
```
не надо Id вручную ставить никогда

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/listener/TelegramBotUpdatesListener.java:
```diff
@@ -303,59 +438,187 @@ private void handleCommand(String command,
+     * Метод по обработке сообщений из чата волонтеров.
+     */
+    private void handleVolunteersCommand(String message, long chatId) {
+        Pattern probationPattern = Pattern.compile(
```
паттерн в константу класса лучше вынести

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/listener/TelegramBotUpdatesListener.java:
```diff
@@ -303,59 +438,187 @@ private void handleCommand(String command,
+    private void handleVolunteersCommand(String message, long chatId) {
+        Pattern probationPattern = Pattern.compile(
+                "^(/\\w+)(/)(\\d+)$"); //паттерн на команду из чата волонтеров с id опекуна
+        Pattern carerIdPattern = Pattern.compile(
```
паттерн в константу класса лучше вынести

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/listener/TelegramBotUpdatesListener.java:
```diff
@@ -386,10 +652,18 @@ private void handleTextMessage(String message, long chatId, long volunteerChatId
-            this.dailyReportService.addDailyReport(dailyReport);
+            Carer carer = this.carerService.findCarerByChatId(chatId, client.getPetType());
+            DailyReport dailyReport = this.dailyReportService.findDailyReportByCarerIdAndDate(carer.getId(), LocalDate.now(), client.getPetType());
+            if (client.getPetType().equals(Pet.CAT)) {
```
enum-ы можно сравнивать через ==, т.к. это константы

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/listener/TelegramBotUpdatesListener.java:
```diff
@@ -386,10 +652,18 @@ private void handleTextMessage(String message, long chatId, long volunteerChatId
+                CatDailyReport catDailyReport = (CatDailyReport) dailyReport;
+                catDailyReport.setCatDiet(message.substring(2).trim());
+                this.dailyReportService.addDailyReport(catDailyReport, client.getPetType());
+            } else if (client.getPetType().equals(Pet.DOG)) {
```
enum-ы можно сравнивать через ==, т.к. это константы

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/listener/TelegramBotUpdatesListener.java:
```diff
@@ -398,10 +672,17 @@ private void handleTextMessage(String message, long chatId, long volunteerChatId
-            this.dailyReportService.addDailyReport(dailyReport);
+            Carer carer = this.carerService.findCarerByChatId(chatId, client.getPetType());
+            DailyReport dailyReport = this.dailyReportService.findDailyReportByCarerIdAndDate(carer.getId(), LocalDate.now(), client.getPetType());
+            if (client.getPetType().equals(Pet.CAT)) {
```
enum-ы можно сравнивать через ==, т.к. это константы

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/listener/TelegramBotUpdatesListener.java:
```diff
@@ -398,10 +672,17 @@ private void handleTextMessage(String message, long chatId, long volunteerChatId
+                CatDailyReport catDailyReport = (CatDailyReport) dailyReport;
+                catDailyReport.setCatHealth(message.substring(2).trim());
+                this.dailyReportService.addDailyReport(catDailyReport, client.getPetType());
+            } else if (client.getPetType().equals(Pet.DOG)) {
```
enum-ы можно сравнивать через ==, т.к. это константы

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/listener/TelegramBotUpdatesListener.java:
```diff
@@ -411,27 +692,48 @@ private void handleTextMessage(String message, long chatId, long volunteerChatId
+                CatDailyReport catDailyReport = (CatDailyReport) dailyReport;
+                catDailyReport.setCatBehavior(message.substring(2).trim());
+                this.dailyReportService.addDailyReport(catDailyReport, client.getPetType());
+            } else if (client.getPetType().equals(Pet.DOG)) {
```
enum-ы можно сравнивать через ==, т.к. это константы

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/listener/TelegramBotUpdatesListener.java:
```diff
@@ -411,27 +692,48 @@ private void handleTextMessage(String message, long chatId, long volunteerChatId
-            this.dailyReportService.addDailyReport(dailyReport);
+            Carer carer = this.carerService.findCarerByChatId(chatId, client.getPetType());
+            DailyReport dailyReport = this.dailyReportService.findDailyReportByCarerIdAndDate(carer.getId(), LocalDate.now(), client.getPetType());
+            if (client.getPetType().equals(Pet.CAT)) {
```
enum-ы можно сравнивать через ==, т.к. это константы

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/listener/TelegramBotUpdatesListener.java:
```diff
@@ -503,38 +807,67 @@ private void shelterInfoCommandMenu(long chatId) {
+    private void takePetCommandMenu(long chatId, String petType) {
+        List<InlineKeyboardButton> buttons = null;
+        String text = "";
+        if (petType.equals(Pet.CAT)) {
```
enum-ы можно сравнивать через ==, т.к. это константы

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/listener/TelegramBotUpdatesListener.java:
```diff
@@ -503,38 +807,67 @@ private void shelterInfoCommandMenu(long chatId) {
+                            .callbackData(Command.BACK_COMMAND.getCommand())
+            ));
+            text = "Добрый день! Здесь Вы можете узнать как взять кошку из приюта.";
+        } else if (petType.equals(Pet.DOG)) {
```
enum-ы можно сравнивать через ==, т.к. это константы

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/listener/TelegramBotUpdatesListener.java:
```diff
@@ -556,8 +889,14 @@ private void sendReportCommandMenu(long chatId) {
-        SendMessage response = new SendMessage(chatId, "Здесь Вы можете узнать " +
-                "как отправить отчёт о питомце.");
+        String text = "";
+        if (petType.equals(Pet.CAT)) {
```
enum-ы можно сравнивать через ==, т.к. это константы

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/listener/TelegramBotUpdatesListener.java:
```diff
@@ -556,8 +889,14 @@ private void sendReportCommandMenu(long chatId) {
+        String text = "";
+        if (petType.equals(Pet.CAT)) {
+            text = "Здесь Вы можете узнать как отправить отчёт о кошке.";
+        } else if (petType.equals(Pet.DOG)) {
```
enum-ы можно сравнивать через ==, т.к. это константы

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/AgreementService.java:
```diff
@@ -17,43 +23,69 @@
-            LOGGER.info("Was invoked method for adding agreement");
-            return modelMapper.mapToAgreementRecord(agreement);
+            Agreement agreement = this.modelMapper.mapToAgreementEntity(agreementRecord);
+            if (agreementRecord.getPetType().equals(Pet.CAT)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/AgreementService.java:
```diff
@@ -17,43 +23,69 @@
+                this.catRepository.save(cat);
+                LOGGER.info("Was invoked method for adding cat agreement");
+                return modelMapper.mapToAgreementRecord(catAgreement);
+            } else if (agreementRecord.getPetType().equals(Pet.DOG)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/AgreementService.java:
```diff
@@ -17,43 +23,69 @@
-                    .orElseThrow(() -> new IllegalArgumentException("Договор не найден"));
-            LOGGER.info("Was invoked method to find agreement by id");
-            return modelMapper.mapToAgreementRecord(agreement);
+            if (petType.equals(Pet.CAT)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/AgreementService.java:
```diff
@@ -17,43 +23,69 @@
+                        .orElseThrow(() -> new IllegalArgumentException("Договор не найден"));
+                LOGGER.info("Was invoked method to find cat agreement by id");
+                return modelMapper.mapToAgreementRecord(catAgreement);
+            } else if (petType.equals(Pet.DOG)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/AgreementService.java:
```diff
@@ -63,55 +95,107 @@ public AgreementRecord findAgreementById(long id) {
-                    .save(modelMapper.mapToAgreementEntity(agreementRecord));
-            return modelMapper.mapToAgreementRecord(agreement);
+            Agreement agreement = this.modelMapper.mapToAgreementEntity(agreementRecord);
+            if (agreementRecord.getPetType().equals(Pet.CAT)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/AgreementService.java:
```diff
@@ -63,55 +95,107 @@ public AgreementRecord findAgreementById(long id) {
+                LOGGER.info("Was invoked method to edit cat agreement");
+                CatAgreement catAgreement = this.catAgreementRepository.save((CatAgreement) agreement);
+                return modelMapper.mapToAgreementRecord(catAgreement);
+            } else if (agreementRecord.getPetType().equals(Pet.DOG)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/AgreementService.java:
```diff
@@ -63,55 +95,107 @@ public AgreementRecord findAgreementById(long id) {
         if (id > 0) {
-            LOGGER.info("Was invoked method to delete agreement");
-            agreementRepository.deleteById(id);
+            if (petType.equals(Pet.CAT)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/AgreementService.java:
```diff
@@ -63,55 +95,107 @@ public AgreementRecord findAgreementById(long id) {
+            if (petType.equals(Pet.CAT)) {
+                LOGGER.info("Was invoked method to delete cat agreement");
+                catAgreementRepository.deleteById(id);
+            } else if (petType.equals(Pet.DOG)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/AgreementService.java:
```diff
@@ -63,55 +95,107 @@ public AgreementRecord findAgreementById(long id) {
-                    .map(this.modelMapper::mapToAgreementRecord)
-                    .toList();
+    public List<AgreementRecord> findAllAgreements(String petType) {
+        if (petType.equals(Pet.CAT)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/AgreementService.java:
```diff
@@ -63,55 +95,107 @@ public AgreementRecord findAgreementById(long id) {
+                LOGGER.info("Was invoked method to find all cat agreements, but cat agreements were not found");
+                return new ArrayList<>();
+            }
+        } else if (petType.equals(Pet.DOG)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/AgreementService.java:
```diff
@@ -63,55 +95,107 @@ public AgreementRecord findAgreementById(long id) {
-                LOGGER.info("Was invoked method to find all agreements with probation for the specified date = " +
-                        localDate);
-                return agreements;
+            if (petType.equals(Pet.CAT)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/AgreementService.java:
```diff
@@ -63,55 +95,107 @@ public AgreementRecord findAgreementById(long id) {
+                            localDate + ", but cat agreements were not found");
+                    return new ArrayList<>();
+                }
+            } else if (petType.equals(Pet.DOG)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/AgreementService.java:
```diff
@@ -120,8 +204,45 @@ public List<Agreement> findAllAgreementsWithProbationByDate(LocalDate localDate)
-        return this.agreementRepository.findAll();
+    @SuppressWarnings("unchecked")
+    public <T extends Agreement> List<T> findAgreementsWithEndingProbation(String petType) {
+        if (petType.equals(Pet.CAT)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/AgreementService.java:
```diff
@@ -120,8 +204,45 @@ public List<Agreement> findAllAgreementsWithProbationByDate(LocalDate localDate)
+            return (List<T>) this.catAgreementRepository.findAll().stream()
+                    .filter(e -> e.getProbationEndData().minusDays(2).equals(LocalDate.now()))
+                    .toList();
+        } else if (petType.equals(Pet.DOG)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/AgreementService.java:
```diff
@@ -120,8 +204,45 @@ public List<Agreement> findAllAgreementsWithProbationByDate(LocalDate localDate)
+
+    public AgreementRecord changeProbationEndData(long id, LocalDate localDate, String petType) {
+        if (id > 0) {
+            if (petType.equals(Pet.CAT)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/AgreementService.java:
```diff
@@ -120,8 +204,45 @@ public List<Agreement> findAllAgreementsWithProbationByDate(LocalDate localDate)
+                catAgreement.setProbationEndData(localDate);
+                this.catAgreementRepository.save(catAgreement);
+                return this.modelMapper.mapToAgreementRecord(catAgreement);
+            } else if (petType.equals(Pet.DOG)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/CarerService.java:
```diff
@@ -18,47 +22,71 @@
-            LOGGER.info("Was invoked method for adding carer from Telegram bot");
-            this.carerRepository.save(carer);
-            return carer;
+            if (petType.equals(Pet.CAT)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/CarerService.java:
```diff
@@ -18,47 +22,71 @@
+                LOGGER.info("Was invoked method for adding cat carer from Telegram bot");
+                this.catCarerRepository.save(catCarer);
+                return catCarer;
+            } else if (petType.equals(Pet.DOG)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/CarerService.java:
```diff
@@ -18,47 +22,71 @@
         if (carer != null) {
-            LOGGER.info("Was invoked method for saving carer from Telegram bot");
-            return this.carerRepository.save(carer);
+            if (petType.equals(Pet.CAT)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/CarerService.java:
```diff
@@ -18,47 +22,71 @@
+            if (petType.equals(Pet.CAT)) {
+                LOGGER.info("Was invoked method for saving cat carer from Telegram bot");
+                return this.catCarerRepository.save((CatCarer) carer);
+            } else if (petType.equals(Pet.DOG)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/CarerService.java:
```diff
@@ -74,15 +102,26 @@ public Carer saveCarer(Carer carer) {
-        if (id < 0) {
+    public CarerRecord findCarer(long id, String petType) {
+        if (id > 0) {
+            if (petType.equals(Pet.CAT)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/CarerService.java:
```diff
@@ -74,15 +102,26 @@ public Carer saveCarer(Carer carer) {
+                CatCarer catCarer = this.catCarerRepository.findById(id).
+                        orElseThrow(() -> new CarerNotFoundException("Опекун с id = " + id + " не найден"));
+                return this.modelMapper.mapToCarerRecord(catCarer);
+            } else if (petType.equals(Pet.DOG)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/CarerService.java:
```diff
@@ -94,11 +133,19 @@ public CarerRecord findCarer(long id) {
-            LOGGER.info("Was invoked method to find carer by chat id");
-            return this.carerRepository.findCarerByChatId(chatId)
-                    .orElse(null);
+            if (petType.equals(Pet.CAT)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/CarerService.java:
```diff
@@ -94,11 +133,19 @@ public CarerRecord findCarer(long id) {
+                LOGGER.info("Was invoked method to find cat carer by chat id from Telegram");
+                return this.catCarerRepository.findCatCarerByChatId(chatId)
+                        .orElse(null);
+            } else if (petType.equals(Pet.DOG)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/CarerService.java:
```diff
@@ -74,15 +102,26 @@ public Carer saveCarer(Carer carer) {
-    public CarerRecord findCarer(long id) {
-        if (id < 0) {
+    public CarerRecord findCarer(long id, String petType) {
+        if (id > 0) {
```
зачем эта проверка? она бесполезна по сути

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/CarerService.java:
```diff
@@ -108,12 +155,24 @@ public Carer findCarerByChatId(long chatId) {
-            this.modelMapper.updateCarer(carerRecord, carer);
-            this.carerRepository.save(carer);
-            return this.modelMapper.mapToCarerRecord(carer);
+            if (carerRecord.getPetType().equals(Pet.CAT)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/CarerService.java:
```diff
@@ -108,12 +155,24 @@ public Carer findCarerByChatId(long chatId) {
     @Transactional
     public CarerRecord editCarer(CarerRecord carerRecord) {
         if (carerRecord != null) {
-            LOGGER.info("Was invoked method to edit carer");
-            Carer carer = this.carerRepository.findById(carerRecord.getId())
-                    .orElseThrow(() -> new CarerNotFoundException("Опекун не найден"));
-            this.modelMapper.updateCarer(carerRecord, carer);
-            this.carerRepository.save(carer);
-            return this.modelMapper.mapToCarerRecord(carer);
+            if (carerRecord.getPetType().equals(Pet.CAT)) {
+                LOGGER.info("Was invoked method to edit cat carer");
+                CatCarer catCarer = this.catCarerRepository.findById(carerRecord.getId())
+                        .orElseThrow(() -> new CarerNotFoundException("Опекун не найден"));
+                this.modelMapper.updateCarer(carerRecord, catCarer);
+                this.catCarerRepository.save(catCarer);
+                return this.modelMapper.mapToCarerRecord(catCarer);
+            } else if (carerRecord.getPetType().equals(Pet.DOG)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/CarerService.java:
```diff
@@ -129,48 +188,76 @@ public CarerRecord editCarer(CarerRecord carerRecord) {
-        if (id < 0) {
+    public void deleteCarer(long id, String petType) {
+        if (id > 0) {
+            if (petType.equals(Pet.CAT)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/CarerService.java:
```diff
@@ -129,48 +188,76 @@ public CarerRecord editCarer(CarerRecord carerRecord) {
+            if (petType.equals(Pet.CAT)) {
+                LOGGER.info("Was invoked method to delete cat carer");
+                this.catCarerRepository.deleteById(id);
+            } else if (petType.equals(Pet.DOG)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/CarerService.java:
```diff
@@ -129,48 +188,76 @@ public CarerRecord editCarer(CarerRecord carerRecord) {
-            Carer carer = this.carerRepository.findCarerByPhoneNumber(phoneNumber)
-                    .orElseThrow(() -> new CarerNotFoundException("Опекун не найден"));
-            return this.modelMapper.mapToCarerRecord(carer);
+            if (petType.equals(Pet.CAT)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/CarerService.java:
```diff
@@ -129,48 +188,76 @@ public CarerRecord editCarer(CarerRecord carerRecord) {
+                CatCarer catCarer = this.catCarerRepository.findCatCarerByPhoneNumber(phoneNumber)
+                        .orElseThrow(() -> new CarerNotFoundException("Опекун не найден"));
+                return this.modelMapper.mapToCarerRecord(catCarer);
+            } else if (petType.equals(Pet.DOG)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/CarerService.java:
```diff
@@ -129,48 +188,76 @@ public CarerRecord editCarer(CarerRecord carerRecord) {
-                    .map(this.modelMapper::mapToCarerRecord)
-                    .toList();
+    public List<CarerRecord> findAllCarers(String petType) {
+        if (petType.equals(Pet.CAT)) {
```
==

dmitry-bizin commented on src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/CarerService.java:
```diff
@@ -129,48 +188,76 @@ public CarerRecord editCarer(CarerRecord carerRecord) {
+                LOGGER.info("Was invoked method to find all cat carers, but carers were not found");
+                return new ArrayList<>();
+            }
+        } else if (petType.equals(Pet.DOG)) {
```
==

ZhosanDenis committed:
Изменения №1 от 27.04.2023
- в контроллерах тип принимаемых параметров с типом животных изменен со String на enum-класс PetType. Также по всем сервисам и TelegramBotUpdatesListener изменен тип данных, показывающих тип животного;
- паттерны сделаны константными полями (TelegramBotUpdatesListener, CarerService);
- скачивание/сохранение в файловую систему сервера фотографии из ежедневного отчета производится через Files.write() - TelegramBotUpdatesListener;
- изменен принцип поиска и сохранения чата волонтеров в телеграм (раньше по id с явным указанием, теперь по типу животного);
- сравнение типов животных в блоках if() осуществляется через ==;
- в таблицы БД clients и volunteer_chats добавлено колонка pet_type.

ZhosanDenis committed:
Merge branch 'dev' into 1-feature-denis

# Conflicts:
#	src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/CarerController.java
#	src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/dto/CarerRecord.java
#	src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/CarerService.java
#	src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/DailyReportService.java
#	src/main/java/pro/sky/teamwork/animalsheltertelegrambotv2/service/DogService.java
#	src/test/java/pro/sky/teamwork/animalsheltertelegrambotv2/controller/DogControllerTest.java

ZhosanDenis committed:
Изменения №2 от 27.04.2023
- удален класс DogService, появившийся в результате слияния.
